lib/turn.rb

```diff
@@ -0,0 +1,68 @@
+class Turn
+    attr_reader :player1, :player2, :spoils_of_war
+    def initialize(player1, player2)
+        @player1 = player1
+        @player2 = player2
+        @spoils_of_war = []
+        @turn_type = nil
+    end
+
+    def type
+        if @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
+            @turn_type = :basic
+        elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0) && @player1.deck.cards.length < 3 || @player2.deck.cards.length < 3
+            @turn_type = :loss
+        elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0) && @player1.deck.rank_of_card_at(2) == @player2.deck.rank_of_card_at(2)
+            @turn_type = :mutually_assured_destruction
+        elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)
+           @turn_type = :war        
+        end
+    end
+
+    def winner
+        if @turn_type == :mutually_assured_destruction
+            return "No Winner"
+        elsif @turn_type == :war
+            return @player1 if @player1.deck.rank_of_card_at(2) > @player2.deck.rank_of_card_at(2)
+            @player2
+        elsif @turn_type == :basic
+            return @player1 if @player1.deck.rank_of_card_at(0) > @player2.deck.rank_of_card_at(0)
+            @player2 
+        elsif @turn_type == :loss
+            return @player1 if @player1.deck.cards.length > @player2.deck.cards.length
+            @player2
+        end
+    end
+
+    def pile_cards
+        if @turn_type == :mutually_assured_destruction
+            players = [@player1, @player2]
+            3.times do
+                players.each do |card|
+                    card.deck.cards.shift
+                end
+            end        
+        elsif @turn_type == :war
+            players = [@player1, @player2]
+            3.times do
+                players.each do |card|
+                    @spoils_of_war << card.deck.cards.shift
+                end
+            end
+        elsif @turn_type == :basic
+            players = [@player1, @player2]
+            players.each do |card|
+                @spoils_of_war << card.deck.cards.shift
+            end
+        end
+    end
+
+    def award_spoils_of_war(winner)
+        if winner == "No Winner"
+            @spoils_of_war = []
+        else
+            (winner.deck.cards << @spoils_of_war.shuffle).flatten!
+            @spoils_of_war = []
+        end
+    end
```

Comment:
In this method, both conditions seem to clear out the `@spoils_of_war` array. To avoid repetition, I think this method could be refactored to look something like this:

```ruby
def award_spoils_of_war(winner)
  if winner != "No Winner"
    (winner.deck.cards << @spoils_of_war.shuffle).flatten! #<== This will only run if there is a winner
  end
  @spoils_of_war = [] #<== This will always run, whether there is a winner or not
end
```

With all that being said, I am wondering why the `@spoils_of_war` needs to be emptied out anyway since it looks like the way you have the game running in your `Game#start` method, a new `turn` object is created for each new turn, so the clearing happens automatically because every new turn starts with an empty `@spoils_of_war`.